File: 2022/05/21/toHex.py
"""
https://leetcode.com/problems/convert-a-number-to-hexadecimal/
Author: Andrew Jarombek
Date: 5/21/2022
"""

def to_hex(num: int) -> str:
    if num == 0:
        return '0'

    if num < 0:
        num = 2 ** 32 + num

    hex = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f']
    result = ""

    while num > 0:
        result = hex[(num % 16)] + result
        num //= 16

    return result

File: 2023/06/17/go/to_hex.go
/**
 * https://leetcode.com/problems/convert-a-number-to-hexadecimal/
 * Author: Andrew Jarombek
 * Date: 6/17/2023
 */

package _go

import "math"

func toHex(num int) string {
	if num == 0 {
		return "0"
	}

	if num < 0 {
		num = int(math.Pow(2, 32)) + num
	}

	result := ""

	for num > 0 {
		val := num % 16

		if val < 10 {
			result = string(rune(val+48)) + result
		} else {
			result = string(rune(val+87)) + result
		}

		num = num / 16
	}

	return result
}
